From b68e6a0b01320b4591a59c1c149fcbe2906a901f Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Sat, 12 Dec 2020 00:18:18 +0200
Subject: [PATCH 18/35] drivers/net/ethernet/ti/cpts.c add trace for cpts
 events

Enable:
CONFIG_FTRACE=y
CONFIG_ENABLE_DEFAULT_TRACERS=y

Commands:
echo console > /sys/kernel/debug/tracing/set_events
echo 0 > /sys/kernel/debug/tracing/trace && sleep 1 && cat /sys/kernel/debug/tracing/trace

Expected output:
 irq/141-4848400-109     [000] .....11 26482.320412: cpts_fifo_read: CPTS_EV: 3 high:04300000 low:16db8400
 irq/141-4848400-109     [000] .....11 26482.341771: cpts_fifo_read: CPTS_EV: 0 high:00000000 low:17325533
 irq/141-4848400-109     [000] .....11 26482.342927: cpts_fifo_read: CPTS_EV: 0 high:00000000 low:1736eda2
 irq/141-4848400-109     [000] .....11 26482.420399: cpts_fifo_read: CPTS_EV: 3 high:04300000 low:1871663d

Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
---
 drivers/net/ethernet/ti/cpts.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/ti/cpts.c b/drivers/net/ethernet/ti/cpts.c
index cc8682ff3833..d0347ab5df4c 100644
--- a/drivers/net/ethernet/ti/cpts.c
+++ b/drivers/net/ethernet/ti/cpts.c
@@ -213,6 +213,8 @@ static int cpts_fifo_read(struct cpts *cpts, int match)
 
 		dev_dbg(cpts->dev, "CPTS_EV: %d high:%08X low:%08x\n",
 			type, event->high, event->low);
+		trace_printk("CPTS_EV: %d high:%08X low:%08x ts:%llu\n",
+			     type, event->high, event->low, event->timestamp);
 		switch (type) {
 		case CPTS_EV_PUSH:
 			WRITE_ONCE(cpts->cur_timestamp, lo);
-- 
2.17.1

